pennylane/gradients/param_shift_hessian.py

```diff
@@ -0,0 +1,320 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for computing the parameter-shift hessian
+of a qubit-based quantum tape.
+"""
+import numpy as np
+
+import pennylane as qml
+
+from pennylane.transforms.tape_expand import expand_invalid_trainable
+from .parameter_shift import _gradient_analysis
+
+
+def _process_gradient_recipe(gradient_recipe, tol=1e-10):
+    """Utility function to process gradient recipes."""
+
+    gradient_recipe = np.array(gradient_recipe).T
+    # remove all small coefficients, shifts, and multipliers
+    gradient_recipe[np.abs(gradient_recipe) < tol] = 0
+    # remove columns where the coefficients are 0
+    gradient_recipe = gradient_recipe[:, :, ~(gradient_recipe[0, 0] == 0)]
+    # sort columns according to abs(shift2) then abs(shift1)
+    gradient_recipe = gradient_recipe[:, :, np.lexsort(np.abs(gradient_recipe)[:, -1])]
```

Comment:
Yes the previous code only had single values to sort, but since we are dealing with pairs here I wanted to ensure [0,0] would show up first in all cases.